sqrl-planner/src/test/java/com/datasqrl/plan/local/analyze/RetailSqrlModule.java

```diff
@@ -0,0 +1,358 @@
+package com.datasqrl.plan.local.analyze;
+
+import static com.datasqrl.name.Name.HIDDEN_PREFIX;
+
+import com.datasqrl.config.SourceFactory;
+import com.datasqrl.error.ErrorCollector;
+import com.datasqrl.io.ExternalDataType;
+import com.datasqrl.io.InMemSourceFactory;
+import com.datasqrl.io.formats.CSVFormat;
+import com.datasqrl.io.formats.JsonLineFormat;
+import com.datasqrl.io.formats.JsonLineFormat.Configuration;
+import com.datasqrl.io.impl.CanonicalizerConfiguration;
+import com.datasqrl.io.impl.inmem.InMemConnector;
+import com.datasqrl.io.tables.TableConfig;
+import com.datasqrl.io.tables.TableSchema;
+import com.datasqrl.io.tables.TableSource;
+import com.datasqrl.io.util.TimeAnnotatedRecord;
+import com.datasqrl.loaders.SqrlModule;
+import com.datasqrl.loaders.TableSourceNamespaceObject;
+import com.datasqrl.name.Name;
+import com.datasqrl.name.NamePath;
+import com.datasqrl.plan.calcite.TypeFactory;
+import com.datasqrl.plan.local.generate.NamespaceObject;
+import com.datasqrl.schema.Multiplicity;
+import com.datasqrl.schema.UniversalTable;
+import com.datasqrl.schema.UniversalTable.Column;
+import com.datasqrl.schema.UniversalTable.ImportFactory;
+import com.datasqrl.schema.constraint.NotNull;
+import com.datasqrl.schema.input.FlexibleFieldSchema.Field;
+import com.datasqrl.schema.input.FlexibleFieldSchema.FieldType;
+import com.datasqrl.schema.input.FlexibleTableSchema;
+import com.datasqrl.schema.input.RelationType;
+import com.datasqrl.schema.input.SchemaElementDescription;
+import com.datasqrl.schema.type.Type;
+import com.datasqrl.schema.type.basic.BooleanType;
+import com.datasqrl.schema.type.basic.DateTimeType;
+import com.datasqrl.schema.type.basic.FloatType;
+import com.datasqrl.schema.type.basic.IntegerType;
+import com.datasqrl.schema.type.basic.StringType;
+import com.datasqrl.schema.type.basic.UuidType;
+import com.google.auto.service.AutoService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+public class RetailSqrlModule implements SqrlModule {
```

Comment:
This class looks strange to me. There is so much code and type mapping in here. Is that really necessary? Is it not possible to use one of the type converters for this?